google/cloud/internal/populate_rest_options.cc

```diff
@@ -35,9 +36,12 @@ Options PopulateRestOptions(Options opts) {
         "https://longrunning.googleapis.com");
   }
   if (opts.has<EndpointOption>()) {
-    std::string endpoint = opts.get<EndpointOption>();
+    auto& endpoint = opts.lookup<EndpointOption>();
+    // Use an unqualified domain name, because we do not seem to reuse
+    // connections with a fully qualified domain name over REST.
+    if (absl::EndsWith(endpoint, ".googleapis.com.")) endpoint.pop_back();
     if (!absl::StartsWithIgnoreCase(endpoint, "http")) {
-      opts.set<EndpointOption>(absl::StrCat("https://", endpoint));
+      endpoint = absl::StrCat("https://", endpoint);
```

Comment:
I think there is an open bug regarding the `StartsWithIgnoreCase(..., "http")`, right?  Can you open one if not?

Reply:
I did not open a bug. I will just send a follow up PR.

Reply:
#13397